src/Console/Command/DispatchCommand.php

```diff
@@ -527,10 +536,103 @@ private function renderFile(Package $package, $repositoryName, $localPath, $dist
                 $stableBranch,
                 $legacyBranch,
                 $branchConfig['docs_path'],
-                str_replace(array(static::PACKAGIST_GROUP.'/', '-bundle'), '', $package->getName()),
-            ), $localContent));
+                str_replace([$this->packagistGroup . '/', '-bundle'], '', $package->getName()),
+            ], $localContent));
         }
         // Restore file permissions after content copy
         $this->fileSystem->chmod($distPath, fileperms($localPath));
     }
+
+    /**
+     * @param string $path
+     * @param array $projectConfig
+     *
+     * @throws \Exception
+     */
+    private function setVersionConstraintsInComposer($path, array $projectConfig, $currentBranch)
+    {
+        $filePath = $path.'/composer.json';
+        if (!$this->fileSystem->exists($filePath)) {
+            throw new \Exception('No composer.json found at: '.$filePath);
+        }
+        $composerAsString = file_get_contents($filePath);
+        $composerAsJson = json_decode($composerAsString, true);
+        if (!isset($composerAsJson['require'])) {
+            throw new \Exception('no require path found in composer.json at: '.$filePath);
+        }
+
+        $branchConfig = $projectConfig['branches'][$currentBranch];
+        $composerAsJson['require']['php'] = $this->evaluateVersionString($branchConfig['php']);
+
+        // set version constraints for symfony dependencies
+        if (isset($branchConfig['versions']['symfony'])) {
+            foreach ($composerAsJson['require'] as $package => $version) {
+                if ('symfony/monolog-bundle' === $package) {
+                    $composerAsJson['require'][$package] = '~3.1';
+                } elseif ('symfony/phpunit-bridge' === $package) {
+                    $composerAsJson['require'][$package] = trim(
+                        $this->evaluateVersionString($branchConfig['versions']['symfony']),
+                        '^2.8 | '
+                    );
+                }  elseif (preg_match('/symfony\//', $package) && 'friendsofsymfony/jsrouting-bundle' !== $package) {
```

Comment:
@dbu you are better in regex, right? can you give me an regex not to exclude `friendsofsymfony/` that ugly way. I should be something that starts with `symfony/`

Reply:
this is working with the parsed json, right? then you could do `preg_match('/^symfony\//', $package)`. the caret symbol at the beginning of the expression matches the beginning of the subject.